cypress/utils/dashboards/data_explorer/commands.js

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { toTestId } from '../constants';
+
+Cypress.Commands.add('verifyTimeConfig', (start, end) => {
+  const opts = { log: false };
+
+  cy.getElementByTestId('superDatePickerstartDatePopoverButton', opts)
+    .should('be.visible')
+    .should('have.text', start);
+
+  cy.getElementByTestId('superDatePickerendDatePopoverButton', opts)
+    .should('be.visible')
+    .should('have.text', end);
+});
+
+Cypress.Commands.add('saveSearch', (name) => {
+  cy.log('in func save search');
+  const opts = { log: false };
+
+  cy.getElementByTestId('discoverSaveButton', opts).click();
+  cy.getElementByTestId('savedObjectTitle').clear().type(name);
+  cy.getElementByTestId('confirmSaveSavedObjectButton').click({ force: true });
+
+  // Wait for page to load
+  cy.waitForLoader();
+});
+
+Cypress.Commands.add('loadSaveSearch', (name) => {
+  const opts = {
+    log: false,
+    force: true,
+  };
+
+  cy.getElementByTestId('discoverOpenButton', opts).click(opts);
+  cy.getElementByTestId(`savedObjectTitle${toTestId(name)}`).click();
+
+  cy.waitForLoader();
+});
+
+Cypress.Commands.add('verifyHitCount', (count) => {
+  cy.getElementByTestId('discoverQueryHits')
+    .should('be.visible')
+    .should('have.text', count);
+});
+
+Cypress.Commands.add('waitForSearch', () => {
+  Cypress.log({
+    name: 'waitForSearch',
+    displayName: 'wait',
+    message: 'search load',
+  });
+
+  cy.getElementByTestId('docTable');
+});
+
+Cypress.Commands.add('prepareTest', (fromTime, toTime, interval) => {
+  cy.setTopNavDate(fromTime, toTime);
+  // wait until the search has been finished
+  cy.waitForSearch();
+  cy.get('select').select(`${interval}`);
+  cy.waitForLoader();
+});
+
+Cypress.Commands.add('isChartCanvasExist', () => {
+  cy.get('.echChart canvas:last-of-type').should('be.visible');
+});
+
+Cypress.Commands.add('isChartIntervalWarningIconExist', () => {
+  cy.waitForLoader();
+  cy.get('.euiToolTipAnchor').should('be.visible');
+});
```

Comment:
Id remove these two commands and move the code to be next to the test itself for readability. They both appear to be used in just one test.

Reply:
done